python/summarize_sgfs.py

```diff
@@ -0,0 +1,295 @@
+          f"search "
+          f"directory {input_file_dir}.")
+
+  def _add_a_game_file_name(self, input_file_name):
+    if not os.path.isfile(input_file_name):
+      print(f"There is no file with name {input_file_name}.")
+      return None
+    if input_file_name.split(".")[-1].lower() == "sgf":
+      self._new_sgf_files = {input_file_name}.difference(self._all_sgf_files)
+      if len(self._new_sgf_files) == 0:
+        print(f"File name {input_file_name} has been added before")
+        return None
+      else:
+        self._all_sgf_files = self._all_sgf_files.union({input_file_name})
+    elif input_file_name.split(".")[-1].lower() == "sgfs":
+      self._new_sgfs_files = {input_file_name}.difference(self._all_sgfs_files)
+      if len(self._new_sgfs_files) == 0:
+        print(f"File name {input_file_name} has been added before")
+        return None
+      else:
+        self._all_sgfs_files = self._all_sgfs_files.union({input_file_name})
+    else:
+      print(f"{input_file_name} is not sgf or sgfs file, no game was added")
+      return None
+
+  def _add_new_games_to_result_dict(self):
+    """add all sgfs files first"""
+    idx = 1
+    for sgfs in self._new_sgfs_files:
+      self._add_one_sgfs_file_to_result(sgfs)
+      if (idx % 10 == 0):
+        print(f"Addedd {idx} files out of {len(self._new_sgfs_files)} sgfs files.")
+      idx += 1
+    print(f"We have added additional {len(self._new_sgfs_files)} sgfs files into the results.")
+
+    idx = 1
+    for sgf in self._new_sgf_files:
+      self._add_one_sgf_file_to_result(sgf)
+      if (idx % 10 == 0):
+        print(f"Added {idx} files out of {len(self._new_sgf_files)} sgf files.")
+      idx += 1
+    print(f"We have added additional {len(self._new_sgf_files)} sgf files into the results.")
+
+  def _add_one_sgfs_file_to_result(self, sgfs_file_name):
+    """Add a single sgfs file. Each row of such sgf file contain a single game as a sgf string"""
+    if not os.path.exists(sgfs_file_name):
+      print(f"There is no SGFs file as {sgfs_file_name}")
+      return None
+
+    with open(sgfs_file_name, "rb") as f:
+      sgfs_strings = f.readlines()
+
+    for sgf in sgfs_strings:
+      self._add_a_single_sgf_string(sgf)
+
+  def _add_one_sgf_file_to_result(self, sgf_file_name):
+    """Add a single sgf file."""
+    if not os.path.exists(sgf_file_name):
+      print(f"There is no SGF file as {sgf_file_name}")
+      return None
+
+    with open(sgf_file_name, "rb") as f:
+      sgf = f.read()
+
+    self._add_a_single_sgf_string(sgf)
+
+  def _add_a_single_sgf_string(self, sgf_string):
+    """add a single game in a sgf string save the results in self.results."""
+    game = sgf.Sgf_game.from_bytes(sgf_string)
+    winner = game.get_winner()
+    pla_black = game.get_player_name('b')
+    pla_white = game.get_player_name('w')
+    if (game.get_handicap() is not None) or game.get_komi() < 5.5 or game.get_komi() > 7.5:
```

Comment:
Probably we should tolerate game.get_handicap() == 0?

Reply:
We parse the sgf string based on sgfmill package. In the source code when handicap is equal to 0, it will return to None. So our case has covered equal to 0 case. 

   ```
 def get_handicap(self):
        ...
        if handicap == 0:
            handicap = None
        elif handicap == 1:
            raise ValueError
        return handicap
```